## Graded Assignment - 2

#### Q10. Provide the 5 number summary for AMT_INCOME_TOTAL i.e. What is min, max, 25%, 75% and 50% (Median) value.(5 mark)


In [1]:
# Importing required supporting libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
# Creating DF object fromn the Application records

df = pd.read_csv('C:\AI-ML\SampleData\Application_record.csv')


In [3]:
df.describe(include='all',percentiles=[0.01,0.25,0.5,0.75,0.99])

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY   CNT_CHILDREN  \
count   4.385570e+05      438557       438557          438557  438557.000000   
unique           NaN           2            2               2            NaN   
top              NaN           F            N               Y            NaN   
freq             NaN      294440       275459          304074            NaN   
mean    6.022176e+06         NaN          NaN             NaN       0.427390   
std     5.716370e+05         NaN          NaN             NaN       0.724882   
min     5.008804e+06         NaN          NaN             NaN       0.000000   
1%      5.024430e+06         NaN          NaN             NaN       0.000000   
25%     5.609375e+06         NaN          NaN             NaN       0.000000   
50%     6.047745e+06         NaN          NaN             NaN       0.000000   
75%     6.456971e+06         NaN          NaN             NaN       1.000000   
99%     7.637181e+06         NaN          NaN             NaN       3.000000   
max     7.999952e+06         NaN          NaN             NaN      19.000000   

        AMT_INCOME_TOTAL NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
count       4.385570e+05           438557                         438557   
unique               NaN                5                              5   
top                  NaN          Working  Secondary / secondary special   
freq                 NaN           226104                         301821   
mean        1.875243e+05              NaN                            NaN   
std         1.100869e+05              NaN                            NaN   
min         2.610000e+04              NaN                            NaN   
1%          5.400000e+04              NaN                            NaN   
25%         1.215000e+05              NaN                            NaN   
50%         1.607805e+05              NaN                            NaN   
75%         2.250000e+05              NaN                            NaN   
99%         5.400000e+05              NaN                            NaN   
max         6.750000e+06              NaN                            NaN   

       NAME_FAMILY_STATUS  NAME_HOUSING_TYPE     DAYS_BIRTH  DAYS_EMPLOYED  \
count              438557             438557  438557.000000  438557.000000   
unique                  5                  6            NaN            NaN   
top               Married  House / apartment            NaN            NaN   
freq               299828             393831            NaN            NaN   
mean                  NaN                NaN  -15997.904649   60563.675328   
std                   NaN                NaN    4185.030007  138767.799647   
min                   NaN                NaN  -25201.000000  -17531.000000   
1%                    NaN                NaN  -24013.000000  -10913.000000   
25%                   NaN                NaN  -19483.000000   -3103.000000   
50%                   NaN                NaN  -15630.000000   -1467.000000   
75%                   NaN                NaN  -12514.000000    -371.000000   
99%                   NaN                NaN   -8575.000000  365243.000000   
max                   NaN                NaN   -7489.000000  365243.000000   

        FLAG_MOBIL  FLAG_WORK_PHONE     FLAG_PHONE     FLAG_EMAIL  \
count     438557.0    438557.000000  438557.000000  438557.000000   
unique         NaN              NaN            NaN            NaN   
top            NaN              NaN            NaN            NaN   
freq           NaN              NaN            NaN            NaN   
mean           1.0         0.206133       0.287771       0.108207   
std            0.0         0.404527       0.452724       0.310642   
min            1.0         0.000000       0.000000       0.000000   
1%             1.0         0.000000       0.000000       0.000000   
25%            1.0         0.000000       0.000000       0.000000   
50%            1.0         0.000000       0.00000

In [4]:
df_mean = df.AMT_INCOME_TOTAL.mean()
print('Mean Income value = {}'.format(df_mean))

df_min = df.AMT_INCOME_TOTAL.min()
print('Minimum Income value = {}'.format(df_min))

df_max = df.AMT_INCOME_TOTAL.max()
print('Maximum Income value = {}'.format(df_max))

df_std = df.AMT_INCOME_TOTAL.std()
print('Standard deviation  = {}'.format(df_std))

df_median = df.AMT_INCOME_TOTAL.median()
print('Median value = {}'.format(df_median))


df_q1 = df.AMT_INCOME_TOTAL.quantile(0.25)
print('Q1 value = {}'.format(df_q1))

df_q2 = df.AMT_INCOME_TOTAL.quantile(0.5)
print('Q2 value = {}'.format(df_q2))

df_q3 = df.AMT_INCOME_TOTAL.quantile(0.75)
print('Q3 value = {}'.format(df_q3))


Mean Income value = 187524.28600950394
Minimum Income value = 26100.0
Maximum Income value = 6750000.0
Standard deviation  = 110086.85306621535
Median value = 160780.5
Q1 value = 121500.0
Q2 value = 160780.5
Q3 value = 225000.0


#### Summary points for AMT_INCOME_TOTAL column
#### 1. The Mean/Average value is =  187524.28600950394
#### 2. The Min value is = 26100.0
#### 3. The Max value is = 6750000.0
#### 4. The 25% value is 121500.0
####      -  This is known as Q1 or 1st quartile
####      -  It value suggests that there are 25% of the data which has value lower tha Q1 value OR 75% of data having value higher than Q1
#### 5. The 50% value is 160780.5
####     -  This is known as Q2 or 2nd quartile
####     -  It value suggests that there are 50% of the data which has value lower than Q2 value
####     -  This is also equal to Median as it split the distribution exactly in half
#### 6. The 75% value is 
####     -  This is known as Q3 or 3rd quartile
####     -  It value suggests that there are 75% of the data which has value lower than Q3 value Or 25% of data having value higher than Q3